OrbitGl/TimeGraph.cpp

```diff
@@ -377,6 +383,88 @@ void TimeGraph::ProcessTimer(const TimerInfo& timer_info) {
+void TimeGraph::ProcessOrbitFunctionTimer(const FunctionInfo* function,
+                                          const TimerInfo& timer_info) {
+  FunctionInfo::OrbitType type = function->type();
+  uint64_t time = timer_info.start();
+  CHECK(timer_info.registers_size() > 0);
+  uint64_t graph_id = timer_info.registers(0);
+
+  switch (type) {
+    case FunctionInfo::kOrbitTrackInt: {
+      int32_t value = static_cast<int32_t>(timer_info.registers(1));
+      auto track = GetOrCreateGraphTrack(graph_id);
+      track->AddValue(value, time);
+    } break;
+    case FunctionInfo::kOrbitTrackInt64: {
+      int64_t value = static_cast<int64_t>(timer_info.registers(1));
+      auto track = GetOrCreateGraphTrack(graph_id);
+      track->AddValue(value, time);
+    } break;
+    case FunctionInfo::kOrbitTrackUint: {
+      uint32_t value = static_cast<uint32_t>(timer_info.registers(1));
+      auto track = GetOrCreateGraphTrack(graph_id);
+      track->AddValue(value, time);
+    } break;
+    case FunctionInfo::kOrbitTrackUint64: {
+      uint64_t value = static_cast<uint64_t>(timer_info.registers(1));
+      auto track = GetOrCreateGraphTrack(graph_id);
+      track->AddValue(value, time);
+    } break;
+    case FunctionInfo::kOrbitTrackFloatAsInt: {
+      int32_t int_value = static_cast<int32_t>(timer_info.registers(1));
+      float value = *(reinterpret_cast<float*>(&int_value));
+      auto track = GetOrCreateGraphTrack(graph_id);
+      track->AddValue(value, time);
+      break;
+    }
+    case FunctionInfo::kOrbitTrackDoubleAsInt64: {
+      int64_t int_value = static_cast<int64_t>(timer_info.registers(1));
+      double value = *(reinterpret_cast<double*>(&int_value));
+      auto track = GetOrCreateGraphTrack(graph_id);
+      track->AddValue(value, time);
+      break;
+    }
+    case FunctionInfo::kOrbitTimerStart:
+      ProcessManualIntrumentationTimer(timer_info);
+      break;
+    default:
+      break;
+  }
+}
+
+void TimeGraph::ProcessManualIntrumentationTimer(const TimerInfo& timer_info) {
+  uint64_t string_address = timer_info.registers(0);
+
+  // Request remote string on first encounter of string_address.
+  if (!manual_instrumentation_string_manager_.Contains(string_address)) {
+    // Prevent redundant string requests while the current request is in flight.
+    manual_instrumentation_string_manager_.AddOrReplace(string_address, "");
+
+    GOrbitApp->GetThreadPool()->Schedule([this, string_address]() {
+      int32_t pid = Capture::GTargetProcess->GetID();
+      const auto& process_manager = GOrbitApp->GetProcessManager();
+      auto error_or_string =
+          process_manager->LoadNullTerminatedString(pid, string_address);
+
+      if (error_or_string.has_value()) {
+        manual_instrumentation_string_manager_.AddOrReplace(
+            string_address, error_or_string.value());
+      } else {
+        ERROR("Error loading remote string %s",
```

Comment:
```suggestion
        ERROR("Loading remote string: %s",
```

("Error" is already printed by the macro.)